### ISSUE TITLE
Search Field is missing prop for aria-label

### ISSUE THREAD

**Issue by JakeLaCombe** [cerner/terra-core#1861]
# Bug Report

## Description
The search field is difficult for screen readers to scan. To improve this aspect, we should provide a prop that populates the aria-label attribute of the input element.

**Comment by bjankord** [comment]
Resolved in https://github.com/cerner/terra-core/pull/1762
